### ISSUE TITLE
Git Extras extension (blame) not showing correct commit author.

### ISSUE THREAD

**Issue by vanesa** [sourcegraph/sourcegraph#19436]
- **Sourcegraph version:** "Sourcegraph.com"

#### Steps to reproduce:

1. Open https://sourcegraph.com/github.com/sourcegraph/about/-/commit/7ee2197408c9a435cf3fc353e9f625d6e7cdb243?visible=30#diff-3e8ea8e732275954c089fcbb83bfa041 with git extras activated. It shows aileenrose as the author of all of the lines. Yet the latest commit of that file was authored by Farhan Attamimi. It's also shown incorrectly VSCode and on GitHub when you open the commit, though that's them showing aileen incorrectly as the last commiter: https://github.com/sourcegraph/about/commit/7ee2197408c9a435cf3fc353e9f625d6e7cdb243
I assume that's the source of the issue?

#### Expected behavior:
https://github.com/sourcegraph/about/pull/2538/files#diff-25d1634d900a5690ddb50b5e372caaec2e7396683a1572a4319d94eda4e84074

![image](https://user-images.githubusercontent.com/9110008/112402014-fbc0da80-8cc8-11eb-830d-beb85326ec9b.png)

#### Actual behavior:

https://github.com/sourcegraph/about/commit/7ee2197408c9a435cf3fc353e9f625d6e7cdb243

![image](https://user-images.githubusercontent.com/9110008/112401717-6291c400-8cc8-11eb-819f-5d8ca25f22a9.png)

In Sourcegraph.com:
![image](https://user-images.githubusercontent.com/9110008/112401809-8c4aeb00-8cc8-11eb-9d2c-eb46de7e6afd.png)

![image](https://user-images.githubusercontent.com/9110008/112401461-d1bae880-8cc7-11eb-93ae-207686ac140b.png)

**Comment by Joelkw** [comment]
@vanesa if Sourcegraph, VSCode, and GitHub are all showing Aileen incorrectly, this makes me think it's not necessarily a bug with our git blame extension (or we all have the same bug) but possibly with the user expectation vs how GitHub or git handle it, since it looks like she opened the original commit. 

Alternatively, it's possible that us (and others) are displaying the first committer when multiple people have edited a line in a commit. But it looks like GitHub is https://github.com/sourcegraph/about/blame/7ee2197408c9a435cf3fc353e9f625d6e7cdb243/website/src/pages/contact/request-batch-changes-demo.tsx to the author of the commit – it could just be this is the standard here?

**Comment by tjkandala** [comment]
To echo Joel, I think this is because we squash commits before merging, so all changes in a PR will be attributed to one person

**Comment by vanesa** [comment]
When looking at a file in Sourcegraph, VSCode and Github I would expect to see the person who wrote/edited the line of code I'm seeing to show up in git blame, not the person who created a PR that later along in the process got that file added. That PR creator never even touched the file I'm looking at. The value that Git Extras (blame) provides is that it points me to the person I can talk to about the code. That value isn't given to me by showing me an unassociated PR maker who wouldn't be able to answer my questions. If Github is sending the wrong data, the extension might have to retrieve that information differently. 

Here it shows the correct code author: https://sourcegraph.com/github.com/sourcegraph/about@11e90917b79ae2bff52811fa89da4bfa5f2ab6d0/-/blob/website/src/pages/contact/request-batch-changes-demo.tsx?utm_source=chrome-extension#L8 
since it's collecting the information from Farhan's specific commit. 

I don't know if the GitHub api allows to retrieve this information from a deep level after the PR is merged, but if not, it unfortunately falsifies the data and defeats the purpose of Git Extras (blame) when the original PR creator isn't the person who authored the code.

**Comment by vanesa** [comment]
Ah, just read @tjkandala comment about squash committing. Yes, that's the reason. I guess it would be too complex to find a workaround for this.

**Comment by muratsu** [comment]
The author seems to be the aileenrose (see the [patch](https://github.com/sourcegraph/about/commit/7ee2197408c9a435cf3fc353e9f625d6e7cdb243.patch)) on git. The reason why we're not seeing Farhan was a bit of a more mystery for me! When I looked further, I realized that commit 11e90917b doesn't exist in the current branch history. I think the branch it existed got orphaned/written over at one point and the commit aileenrose made is considered the first commit with this particular file (see the warning message on [farhans commit](https://github.com/sourcegraph/about/commit/55f084bacac2eea945aa5dbb5b5be137915cd8ff)). I am not entirely sure how github is able to connect the dots but it seems beyond the scope of this extension. I'll keep this issue open but will move this item to the icebox for future reference.
